src/routing/raptor/optimize_footpaths.cc

```diff
@@ -0,0 +1,234 @@
+#include "nigiri/routing/raptor/reconstruct.h"
+
+#include "nigiri/routing/for_each_meta.h"
+#include "nigiri/routing/journey.h"
+#include "nigiri/rt/frun.h"
+#include "nigiri/rt/rt_timetable.h"
+#include "nigiri/timetable.h"
+
+namespace nigiri::routing {
+
+void optimize_start(timetable const& tt,
```

Comment:
Does this consider that `start` in backward search means the last leg of the journey (-> arrival times are relevant)?

Reply:
Good point. So, the legs of the journey are in the order in which a real-world traveller would make the journey. However, if a backward search was used to answer the query, the offsets at the real-world start are in `q.destination_` of the query, right?